### DISCUSSION TITLE
How to get the application window size in WinUI platform?

### DISCUSSION THREAD

**Discussion by karthikraja-arumugam** [dotnet/maui#10897] [General]
In my app I measure my custom layout using parent view's size constraints, but while the parent is VerticalStackLayout or ScrollView the size values are infinity in such cases I had to calculate the size of my layout.

So I'm using `DeviceDisplay.Current.MainDisplayInfo` details but in the windows platform, it returns full screen size. I need only the size of my application window. Is there any approach to get the application window size?

**Comment by xtuzy** [comment]
Use Winui library like https://github.com/dotMorten/WinUIEx to get windows size, or you can try get Page size, page size always not infinity.
But your requirement not need it, when your custom layout's parent is VerticalStackLayout or ScrollView, your layout should measure size according to it child, or do nothing,  you know VerticalStackLayout or ScrollView can have infinity size, they measure size according to child, so you need do it like they.
